datamodel/odata/odata-core/src/main/java/com/sap/cloud/sdk/datamodel/odata/helper/ODataEntitySerializer.java

```diff
@@ -122,6 +123,143 @@ static String serializeEntityForUpdatePatch(
         return GSON_SERIALIZING_NULLS.toJson(partialEntity);
     }
 
+    /**
+     * Serializes an entity for update request (PATCH) including changes in nested properties. Allowing null values.
+     * Resulting JSON contains the full value of complex fields for changing any nested field.
+     *
+     * @param entity
+     *            The OData V2 entity reference.
+     * @param includedFields
+     *            Collection of fields to be included in the update (PATCH) request.
+     * @return The serialized JSON string for entity update request.
+     */
+    @Nonnull
+    static String serializeEntityForUpdatePatchRecursiveFull(
+        @Nonnull final VdmEntity<?> entity,
+        @Nonnull final Collection<FieldReference> includedFields )
+    {
+        final JsonObject fullEntityJson = GSON_SERIALIZING_NULLS.toJsonTree(entity).getAsJsonObject();
+        final JsonObject patchObject = new JsonObject();
+
+        final Set<String> changedFieldNames = new HashSet<>(entity.getChangedFields().keySet());
+        includedFields.stream().map(FieldReference::getFieldName).forEach(changedFieldNames::add);
+        changedFieldNames.forEach(key -> patchObject.add(key, fullEntityJson.get(key)));
+
+        entity
+            .toMapOfFields()
+            .entrySet()
+            .stream()
+            .filter(entry -> !patchObject.has(entry.getKey()))
+            .filter(entry -> entry.getValue() instanceof VdmComplex<?>)
+            .filter(entry -> containsNestedChangedFields((VdmComplex<?>) entry.getValue()))
+            .forEach(entry -> patchObject.add(entry.getKey(), fullEntityJson.get(entry.getKey())));
+
+        log.debug("The following object is serialized for update : {}.", patchObject);
+
+        return GSON_SERIALIZING_NULLS.toJson(patchObject);
+    }
+
+    /**
+     * Checks if the given complex object contains any changed fields in its nested fields.
+     *
+     * @param vdmComplex
+     *            the complex object to check
+     * @return true if the complex object contains any changed fields, false otherwise
+     */
+    private static boolean containsNestedChangedFields( final VdmComplex<?> vdmComplex )
+    {
+        if( !vdmComplex.getChangedFields().isEmpty() ) {
+            return true;
+        }
+
+        return vdmComplex
+            .toMapOfFields()
+            .values()
+            .stream()
+            .filter(complexField -> complexField instanceof VdmComplex<?>)
+            .map(complexField -> (VdmComplex<?>) complexField)
+            .anyMatch(ODataEntitySerializer::containsNestedChangedFields);
+    }
```

Comment:
(Comment)

You could also loosen the argument constraint here and save the type-checks in the consuming method.

```suggestion
    private static boolean containsNestedChangedFields( final Object obj )
    {
        if( obj instanceof VdmComplex<?> vdmComplex ) {
            if( !vdmComplex.getChangedFields().isEmpty() ) {
                return true;
            }
            for( Object o : vdmComplex.toMapOfFields().values() ) {
                if( containsNestedChangedFields(o) ) {
                    return true;
                }
            }
        }
        return false;
    }
```